### ISSUE TITLE
[PTNext Beta07] Strange geometric shadow artifacts when indoors

### ISSUE THREAD

**Issue by GarrryValentine** [sammilucia/cyberpunk-ultra-plus#17]
Images attached show comparison with PT21

1440p//DLSSTweaks custom balanced 0.63 scale//RR//PTNext Insane samples

When in exteriors no issues occur. However, whenever entering an indoor or area with a ceiling, these artifacts appear on the upper third of the screen. They remain on the top of the screen - so they disappear when I look at the ceiling.
They disappear when I switch to PT21.
 
![Cyberpunk2077 2024-04-14 12-13-06](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/cc7faf8d-a710-4fb6-a7bc-3461f2071513)
![Cyberpunk2077 2024-04-14 12-13-12](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/d29d8ed4-7586-46bf-b54e-6854d34cedfc)
![Cyberpunk2077 2024-04-14 12-13-19](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/5d53f059-3605-47c7-8c9c-2973dc8d1473)
![Cyberpunk2077 2024-04-14 12-13-58](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/d6b65679-447f-4e35-9a5a-3064d055c366)
![Cyberpunk2077 2024-04-14 12-14-04](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/6c4abe12-2cae-4ca9-b0b0-706d3834ce3b)

**Comment by GarrryValentine** [comment]
Update from more testing:

If I load a save that is already within an interior, no artifacts appear.

![Cyberpunk2077 2024-04-14 12-44-49](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/796073fc-00bd-47ef-9cc8-0a55191a82d4)
![Cyberpunk2077 2024-04-14 12-44-54](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/2ea96226-47f9-4082-b7b0-577c93ceaa75)
![Cyberpunk2077 2024-04-14 12-46-16](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/167017472/03aa6e1e-3e3a-47a5-a9c3-df5ee0f26f7b)

**Comment by JaidynBelbin** [comment]
Yep, I've had this happen too, usually on the first load. Happened outdoors the other day as well. It's like the shadow gets attached to the camera and just follows it around. Reloading the save or switching to PT21 always fixes it.

**Comment by JaidynBelbin** [comment]
Captured a video of what I experience outdoors, hopefully you can see it (sorry for the long load at the end!) https://drive.google.com/uc?id=1k1lqg0b-OrrAoSAHwGrQrLylJDhO-dMy&export=download

Hope that's helpful in some way. I'm trying to remember, but it seems to only occur after a new install of UltraPlus? Goes away after a few reloads.

**Comment by erbkaiser** [comment]
I am noticing the same.  I am running 4.0-rc1 in the PT Next mode with ray generation active.

Certain lighting conditions cause this weird shadow to appear. This is noticeable on some containers and in Jackie's garage during the Offrenda quest.

If I toggle the debug option 'Enable separate denoiser' to off this shadow disappears. Disabling the denoiser from the 'main' page does not do this. All other mod options are default for PT Next.

Chosen main settings:
PT Next
Insane Quality
16GB VRAM

**Comment by sammilucia** [comment]
> Captured a video of what I experience outdoors, hopefully you can see it (sorry for the long load at the end!) https://drive.google.com/uc?id=1k1lqg0b-OrrAoSAHwGrQrLylJDhO-dMy&export=download

ty. i can't access this video 😊

@erbkaiser hmm, `Enable Separate Denoiser` on the debug page is exactly the same as on the front page 🤔. just to check you don't mean `Enable DI Denoising` or `Enable Custom Denoiser` ?

**Comment by erbkaiser** [comment]
Sorry, I was busy looking for the cause in the debug options and I didn't notice it was the same option.
It is indeed the RTXDI denoising from the main page. Disabling that when the bug occurs instantly clears it.

**Comment by erbkaiser** [comment]
![enable separate denoiser](https://github.com/sammilucia/cyberpunk-ultra-plus/assets/6014923/81a1a9db-6990-4cb3-8aea-14d6713ea246)
This is what I was seeing outdoors, for the record.
